bootcampspringboot/src/main/java/com/max/bootcampspringboot/data/repository/TeamRepository.java

```diff
@@ -14,5 +14,25 @@ public interface TeamRepository extends JpaRepository<Team, Integer> {
     @Query("SELECT t.name, COUNT(e) FROM Team t JOIN t.employees e GROUP BY t.name ORDER BY COUNT(e) DESC LIMIT 1")
     public String findLargestTeam();
 
+    @Query(value = "SELECT team_name,\n" +
+            "         employee_name,\n" +
+            "         skill_name,\n" +
+            "         max_experienceLevel\n" +
+            "FROM \n" +
+            "    (SELECT t.name AS team_name,\n" +
+            "         CONCAT(e.salutation,\n" +
+            "         ' ', e.firstname, ' ', e.lastname) AS employee_name, s.name AS skill_name, k.experienceLevel AS max_experienceLevel, ROW_NUMBER()\n" +
+            "        OVER (PARTITION BY t.name, s.name\n" +
+            "    ORDER BY  k.experienceLevel DESC) AS rn\n" +
+            "    FROM teams t\n" +
+            "    JOIN employees e\n" +
+            "        ON t.id = e.fk_team\n" +
+            "    JOIN knowledges k\n" +
+            "        ON e.id = k.fk_employee\n" +
+            "    JOIN skills s\n" +
+            "        ON k.fk_skill = s.id ) subquery\n" +
+            "WHERE rn = 1\n" +
+            "ORDER BY  team_name", nativeQuery = true)
```

Comment:
Native Query, sollte nich true sein.
Damit umgehst du die Vorteile von Jpa. Jpa ermöglicht dir ja jeden Dialekt anzusprechen ohne das du alle können musst, deswegen wenn du hier native true machst, geht diese Query nur mit postgres

Reply:
höhö war da lazy

Reply:
hab das jetzt in Java Code gelöst